src/axom/core/execution/internal/cuda_exec.hpp

```diff
@@ -55,6 +55,7 @@ struct execution_space< CUDA_EXEC< BLOCK_SIZE, SYNCHRONOUS > >
 
   static constexpr bool async() noexcept { return false; };
   static constexpr bool valid() noexcept { return true; };
+  static constexpr bool onDevice() noexcept { return true; };
```

Comment:
Note, the `cuda_exec.hpp` file consists of two `execution_space` specializations: 

1. ``axom::execution_space< CUDA_EXEC< BLOCK_SIZE, SYNCHRONOUS > >`` , and 
2. ``axom::execution_space< CUDA_EXEC< BLOCK_SIZE, ASYNC > >``

It looks like you only implemented the ``onDevice()`` method for ``CUDA_EXEC< BLOCK_SIZE, SYNCHRONOUS >``. 

Please add it for both. 

In addition, please extend our unit test accordingly in ``core/tests/core_execution_space.cpp``, which is designed to catch this kind of inconsistencies